ui/raidboss/data/05-shb/raid/e12s.js

```diff
@@ -440,6 +468,98 @@ export default {
+          const statueId = parseInt(matches.targetId, 16);
+          const statueData = await window.callOverlayHandler({
+            call: 'getCombatants',
+            ids: [statueId],
+          });
+
+          if (statueData === null || !statueData.combatants || !statueData.combatants.length)
+            return;
+
+          // Center = 0, -75
+          const x = statueData.combatants[0].PosX;
+          const y = statueData.combatants[0].PosY - -75;
+          statue.dist = Math.hypot(x, y);
+          if (data.statues.length > 1) {
+            // Sort so that closest statue (inner) is first
+            data.statues.sort((a, b) => a.dist - b.dist);
+          }
+        }
+      },
+      durationSeconds: 20,
+      infoText: (data, matches, output) => {
+        // Player is the source of the tether for this one, oddly enough
+        if (data.me !== matches.source)
+          return;
+
+        if (data.statues.length > 0 && data.statues[0].id === matches.targetId) {
+          data.statueDir = output.inner();
+        } else if (data.statues.length > 1 && data.statues[1].id === matches.targetId) {
+          data.statueDir = output.outer();
+        } else {
+          data.statueDir = '???';
+        }
+        return output.statue({ statueDir: data.statueDir });
+      },
+      outputStrings: {
+        statue: {
+          en: '${statueDir} Statue',
+        },
+        inner: {
+          en: 'Inner',
+        },
+        outer: {
+          en: 'Outer',
+        }
+      },
+    },
+    {
+      id: 'E12S Promise Palm Of Tempeance SE',
+      netRegex: NetRegexes.startsUsing({ source: 'Guardian Of Eden', id: '58B4' }),
+      durationSeconds: 10,
+      infoText: 'Knockback SE',
+    },
+    {
+      id: 'E12S Promise Palm Of Tempeance SW',
+      netRegex: NetRegexes.startsUsing({ source: 'Guardian Of Eden', id: '58B5' }),
+      durationSeconds: 10,
+      infoText: 'Knockback SW',
+    },
+    {
+      // Time the first dodge for lasers from the left/right split
+      id: 'E12S Promise Statue Laser First',
+      netRegex: NetRegexes.startsUsing({ source: 'Eden\'s Promise', id: '58A[DE]' }),
+      netRegexDe: NetRegexes.startsUsing({ source: 'Edens Verheißung', id: '58A[DE]', capture: false }),
+      netRegexFr: NetRegexes.startsUsing({ source: 'Promesse D\'Éden', id: '58A[DE]', capture: false }),
+      netRegexJa: NetRegexes.startsUsing({ source: 'プロミス・オブ・エデン', id: '58A[DE]', capture: false }),
+      condition: (data) => data.statueDir !== undefined && (!data.statueLaserCount || data.statueLaserCount <= 2),
+      durationSeconds: 3,
+      response: statueBaitResponse,
+      outputStrings: statueBaitOutputStrings,
+    },
+    {
+      // Time the other dodges from the previous lasers going off
+      id: 'E12S Promise Statue 2nd/3rd/4th Laser',
+      netRegex: NetRegexes.ability({ source: 'Chiseled Sculpture', id: '58B3' }),
+      // Extra condition here to handle if multiple people get clipped by the lasers
```

Comment:
I've used suppressSeconds successfully for oopsy to time lasers if that's more straightforward?

Reply:
I tried `suppressSeconds` but it wasn't cooperating. I'm not sure if that's due to some issue on my end, some issue with raidemulator, or if it just doesn't work for this setup, and I gave up on debugging it after 20 minutes or so, just wanted to get the trigger to work.

Reply:
I think `suppressSeconds` is a bug in raidemulator, as per https://github.com/quisquous/cactbot/issues/2161#issuecomment-743408063.

Reply:
I actually had a fix for suppressSeconds but forgot to include it in the patch file, whoops. Updated the first PR comment to include that patch as well. This has been addressed in the latest commit.